plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/actions/CodeWhispererActionPromoter.kt

```diff
@@ -7,23 +7,30 @@ import com.intellij.codeInsight.lookup.impl.actions.ChooseItemAction
 import com.intellij.openapi.actionSystem.ActionPromoter
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.DataContext
-import com.intellij.openapi.editor.actionSystem.EditorAction
-import software.aws.toolkits.jetbrains.services.amazonq.CodeWhispererFeatureConfigService
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupLeftArrowHandler
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupRightArrowHandler
-import software.aws.toolkits.jetbrains.services.codewhisperer.popup.handlers.CodeWhispererPopupTabHandler
+import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus
 import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatusNew
+import software.aws.toolkits.jetbrains.settings.CodeWhispererSettings
 
 class CodeWhispererActionPromoter : ActionPromoter {
     override fun promote(actions: MutableList<out AnAction>, context: DataContext): MutableList<AnAction> {
-        if (CodeWhispererFeatureConfigService.getInstance().getNewAutoTriggerUX()) {
-            val results = actions.toMutableList()
-            if (!CodeWhispererInvocationStatusNew.getInstance().isDisplaySessionActive()) return results
+        val results = actions.toMutableList()
+        if (!CodeWhispererInvocationStatusNew.getInstance().isDisplaySessionActive() &&
+            !CodeWhispererInvocationStatus.getInstance().isDisplaySessionActive()
+        ) {
+            return results
+        }
 
-            results.sortWith { a, b ->
-                if (isCodeWhispererForceAction(a)) {
+        results.sortWith { a, b ->
+            if (isCodeWhispererForceAction(a)) {
+                return@sortWith -1
+            } else if (isCodeWhispererForceAction(b)) {
+                return@sortWith 1
+            }
+
+            if (CodeWhispererSettings.getInstance().isQPrioritizedForTabAccept()) {
+                if (isCodeWhispererAcceptAction(a)) {
                     return@sortWith -1
-                } else if (isCodeWhispererForceAction(b)) {
+                } else if (isCodeWhispererAcceptAction(b)) {
```

Comment:
can we extract this sorting logic to a `Comparator<T>`
and  we can have a simple test case for it

Reply:
it's fine we don't need a dedicate comparator for this guy
```
results.sortWith { a, b ->
            if (isCodeWhispererNavigateAction(a)) {
                return@sortWith -1
            } else if (isCodeWhispererNavigateAction(b)) {
                return@sortWith 1
```

Reply:
```
results.sortWith { a, b ->
            if (isCodeWhispererNavigateAction(a)) {
                return@sortWith -1
            } else if (isCodeWhispererNavigateAction(b)) {
                return@sortWith 1
```

This is just to make sure when popup is active, alt + ] is guranteed to trigger navigations in case they are binded to other key shortcuts.